File: csharp/剑指09用两个栈实现队列.cs

public class CQueue
{
    private int top1, top2;
    private int[] st1,st2;
    public CQueue()
    {
        top1 = top2 = -1;
        st1 = new int[10000];
        st2 = new int[10000];
    }

    public void AppendTail(int value)
    {
        st1[++top1] = value;
    }

    public int DeleteHead()
    {
        if(top2==-1)
        {
            while (top1!=-1)
            {
                st2[++top2] = st1[top1--];
            }
        }

        if(top2==-1)
            return -1;
        top2--;
        return st2[top2 + 1];

    }
}

File: python/main/剑指09用两个栈实现队列.py

class CQueue:

    def __init__(self):
        
        self.st1=[]
        self.st2=[]

    def appendTail(self, value: int) -> None:
        self.st1.append(value)

    def deleteHead(self) -> int:
        if(len(self.st2)):
            return self.st2.pop()
        
        while(len(self.st1)):
            self.st2.append(self.st1.pop())
        
        
        return self.st2.pop() if len(self.st2) else -1